### ISSUE TITLE
Responses at zero solute / zero concentration

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/molarity#199]
From #198 @terracoda said

I found one issue - one edge case which may not have been documented in the design doc, even (sorry). The case is when the user uses home and end keys on the Solute Amount slider and jumps to zero solute from a state where the solution was previously saturated. The solution is described "lighter" which is kind of true, but it might be more appropriate to say it is clear and that it is water. 

**The responses in question** - when jumping from a saturated solution directly to pure water:
- "No longer saturated. Solution lighter, zero concentration." (when Solution Values not checked)
- "No longer saturated at 0.000 molar. Solution lighter." (when Solution Values is checked)

**For comparison the response** - when jumping from a concentrated solution - NOT yet saturated - to water:**
- "Zero concentration. Solution pure water."
- "Zero concentration. Solution pure water."

**Possible Solutions:**
**Option 1:** - Combine "No longer saturated." with our current pure-water phrase to explicitly make it clear that the solution is both no longer saturated and now pure water.
- No longer saturated. Zero concentration. Solution pure water.
- No longer saturated. Zero concentration. Solution pure water.

OR

**Option 2:** Use our pure-water phrase and assume learners will know/deduce this state is no longer saturated.
- Zero concentration. Solution pure water.
- Zero concentration. Solution pure water.

@Matthew-phet, I need your thoughts here.

**Comment by jessegreenberg** [comment]
@terracoda said

@Matthew-phet , do you think we should leave as-is, or got with option 1 or 2 if you think we sould change the response.

**Comment by jessegreenberg** [comment]
@terracoda said

> Just adding 2 screen shots from the A11y View showing the responses in question:

![](https://user-images.githubusercontent.com/1902116/71185922-e30c7e00-2256-11ea-9b88-82520cd214e0.png)


![](https://user-images.githubusercontent.com/1902116/71185932-e7389b80-2256-11ea-8971-8161a3f0f82a.png)

**Comment by jessegreenberg** [comment]
Found while reviewing RC in https://github.com/phetsims/QA/issues/464

**Comment by Matthew-phet** [comment]
I think option 1 is a good compromise. Good catch!

**Comment by jessegreenberg** [comment]
@twant assigning to you to review, does this seem straight forward or would this require intrusive changes to current code?

**Comment by twant** [comment]
I think this can be implemented relatively easily -- I'll take a look this week and let you know if it will be more intrusive/involved than I thought!

**Comment by twant** [comment]
@Matthew-phet @terracoda just to clarify, you think it should say: 
- No longer saturated. Zero concentration. Solution pure water.
And this should apply for both "Solution Values" checked and unchecked. Is that correct?

**Comment by terracoda** [comment]
Yes, that is right. The same for both with values and without values shown. Glad it is easy to do.

That said, @twant, if you run into any trouble with implementing this special response, I think it is safe to go with option 2 and have the same response in all cases when the solution changes from "any amount of solute" to zero solute.

**Comment by twant** [comment]
Okay, committed this change to master. @terracoda do you mind reviewing to see how that sounds to you?

**Comment by terracoda** [comment]
@twant, apologies for not getting to this on the weekend.
I just tested from Master, and the new response is good when jumping the Solute Amount slider from a concentrated state to zero solute.

However, if I then move focus to the Solution Volume slider, and adjust Volume when there is zero solute, I also get the new response, which is not correct.

@twant, the new response should never show up when adjusting Solution Volume because I can never jump from a concentrated state to pure water (no solute) when adjusting Solution Volume.

This new response can only actually happen when jumping the Solute Amount slider.
Apologies for not making that clear.

The repeated "new responses" below are happening on the Solution Volume Slider - minus the first one , of course/

![Screen Shot 2019-12-23 at 10 32 57 AM](https://user-images.githubusercontent.com/1902116/71362282-33921d00-2570-11ea-92e4-fd386146503e.png)
![Screen Shot 2019-12-23 at 10 34 37 AM](https://user-images.githubusercontent.com/1902116/71362294-3bea5800-2570-11ea-870e-fe4fd95fd13f.png)

**Comment by terracoda** [comment]
When adjusting Solution Volume when there is no solute, the response should be:
- "Zero concentration. Solution pure water."

**Comment by terracoda** [comment]
And, we can always go with option 2 in https://github.com/phetsims/molarity/issues/199#issue-540387499 if option 1 is not possible in our tight timeline.

**Comment by twant** [comment]
Great catch, @terracoda! I'm pushing a change that I think fixes this issue -- let me know how it looks to you!

**Comment by terracoda** [comment]
@twant, you fixed the case I mentioned above - great!
Unfortunately, when using the blue solution, I didn't notice what happens if from the solution-darkest state to zero solute.

You can do this easily with the purple solution - see alert underlined in read. I turned on values to make it easier to reproduce. 

I think the new alert is technically correct, but we didn’t say “saturated” in the previous state, so i am unsure if we should say "No longer saturated" when we jump to zero solute.

![Screen Shot 2019-12-23 at 5 39 55 PM](https://user-images.githubusercontent.com/1902116/71381011-5510fa00-25ac-11ea-8622-645a3d5bc702.png)

How does the model view this "solution-darkest" state - the equilibrium state?
Is it easy to eliminate this case as well?

**Comment by terracoda** [comment]
@twant and I had a zoom call on this, and we think this case along with another one that @twant found is fixed.

**Comment by twant** [comment]
Just updated per a zoom call with @terracoda, where we reviewed these changes together. I'm going to close this issue, but please feel free to reopen, @terracoda if you think it needs additional review!

**Comment by zepumph** [comment]
There may be commits here to cherry pick into the next rc.

**Comment by jessegreenberg** [comment]
@terracoda can you please verify that this is resolved in https://phet-dev.colorado.edu/html/molarity/1.5.0-rc.3/phet/molarity_all_phet.html

(a11y view) https://phet-dev.colorado.edu/html/molarity/1.5.0-rc.3/phet/molarity_a11y_view.html

**Comment by terracoda** [comment]
@jessegreenberg, the context responses around zero concentration are working perfectly.

If I jump from the "solution darkest" state (saturated with no precipitate) to zero solute, I get:
"no potassium dichromate" then "Zero concentration. Solution pure water."

This is perfect because we describe being saturated in the previous sate.

If I jump from a described saturated state (i.e. saturated and showing precipitate) to no solute, I get a  more explicit response:
"No longer saturated. Zero concentration. Solution pure water."

This is all working perfectly.

**Comment by jessegreenberg** [comment]
Great, thanks for checking.  Closing since this has successfully moved into the RC branch.
